cmd/loadtest/server.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-load-test-ng/config"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control/simplecontroller"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/store/memstore"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user/userentity"
+	"github.com/mattermost/mattermost-server/mlog"
+	"github.com/spf13/cobra"
+)
+
+type Agent struct {
+	loadTests map[string]*loadtest.LoadTester
+}
+
+func (a *Agent) createLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	var config config.LoadTestConfig
+	err := json.NewDecoder(r.Body).Decode(&config)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	newSimpleController := func(id int, status chan<- control.UserStatus) control.UserController {
+		ueConfig := userentity.Config{
+			ServerURL:    config.ConnectionConfiguration.ServerURL,
+			WebSocketURL: config.ConnectionConfiguration.WebSocketURL,
+		}
+		ue := userentity.New(memstore.New(), ueConfig)
+		return simplecontroller.New(id, ue, status)
+	}
+
+	lt := loadtest.New(&config, newSimpleController)
+	a.loadTests[lt.Id] = lt
+	// start := time.Now()
+	// err = lt.Run()
+	// if err != nil {
+	// 	return err
+	// }
+
+	// mlog.Info("loadtest started")
+	// time.Sleep(60 * time.Second)
+
+	// err = lt.Stop()
+	// mlog.Info("loadtest done", mlog.String("elapsed", time.Since(start).String()))
+
+	// return err
+
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]string{"loadTestId": lt.Id})
+
+}
+
+func (a *Agent) getLoadTestById(w http.ResponseWriter, r *http.Request) (*loadtest.LoadTester, error) {
+	vars := mux.Vars(r)
+	id := vars["id"]
+	lt, ok := a.loadTests[id]
+	if !ok {
+		err := fmt.Errorf("Load test with id %s not found", id)
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return nil, err
+	}
+	return lt, nil
+}
+func (a *Agent) runLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	go lt.Run()
```

Comment:
Run is not a blocking call. It will return rather quickly. No need to create an additional goroutine.